### PR TITLE
Add WhatsYourSign 1.2.1

### PR DESCRIPTION
**Pull request by vitorgalvao**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked that the cask was not already refused in [closed issues].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed

### CODE REVIEW

File: Casks/whatsyoursign.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cask 'whatsyoursign' do
+  version '1.2.1'
+  sha256 'b05363ce2106e23eda95caef1f812fb6cb1dc8f983dee035a4730afe35fc13d6'
+
+  # bitbucket.org/objective-see was verified as official when first introduced to the cask
+  url "https://bitbucket.org/objective-see/deploy/downloads/WhatsYourSign_#{version}.zip"
+  appcast 'https://objective-see.com/products.json',
+          checkpoint: 'b2004932186e0eb176a9ac01fe499de292d502d7a17559e900822f9cdbfa74d6'
+  name 'What\'s YourSign?'
```
Comment (miccal) [inline Casks/whatsyoursign.rb]: Should this be `name 'What\'s Your Sign?'`?

File: Casks/whatsyoursign.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  homepage 'https://objective-see.com/products/whatsyoursign.html'
+
+  installer manual: 'WhatsYourSign_Installer.app'
+
+  uninstall_postflight do
+    system_command '/usr/bin/pluginkit', args: ['-r', '-i', 'com.objective-see.WhatsYourSignExt.FinderSync']
```
Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: I’m not really sure this works. It **should**, but I have no VM to test on (Veertu is months late on their macOS guests release and my VMWare hangs the system).

Comment (miccal) [inline Casks/whatsyoursign.rb]: I have a testing machine, would you like me to test it? What would you like me to do exactly?

File: Casks/whatsyoursign.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: Yes. It should work fine.

Comment (miccal) [inline Casks/whatsyoursign.rb]: No, I mean should there be a space between `Your` and `Sign`?

File: Casks/whatsyoursign.rb (hunk 2 above)
Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: Thank you. To test, install it first (with the manual installer).

Then
```shell
# run this
find ~/Library -iname 'com.objective-see.WhatsYourSignExt.FinderSync'
# you should get
~/Application Scripts/com.objective-see.WhatsYourSignExt.FinderSync
~/Containers/com.objective-see.WhatsYourSignExt.FinderSync` # and some other inside this one
```

Also verify in `WhatsYourSign` exists in `System Preferences → Extensions → Finder`.

Now run that uninstall command (`/usr/bin/pluginkit -r -i com.objective-see.WhatsYourSignExt.FinderSync`) and see what it outputs. Finally, check if the aforementioned things were removed. If the command fails, perhaps try with the full path (`/usr/bin/pluginkit -r -i ~/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync`).

File: Casks/whatsyoursign.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: Ah, yes! Yes, it should.

File: Casks/whatsyoursign.rb (hunk 2 above)
Comment (miccal) [inline Casks/whatsyoursign.rb]: Installing manually, then running `sudo find / -iname '*objective-see*'` (to be thorough) reported nothing. After reboot, same command found the following:
```
/Users/raesy/Library/Application Scripts/com.objective-see.WhatsYourSignExt.FinderSync
/Users/raesy/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync
/Users/raesy/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync/Data/Library/Application Scripts/com.objective-see.WhatsYourSignExt.FinderSync
/Users/raesy/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync/Data/Library/Preferences/com.objective-see.WhatsYourSignExt.FinderSync.LSSharedFileList.plist
```

Verified that **WhatsYourSign** exists in **System Preferences → Extensions** (both before and after reboot).

Running the commands `/usr/bin/pluginkit -r -i com.objective-see.WhatsYourSignExt.FinderSync` and `/usr/bin/pluginkit -r -i ~/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync` output nothing.

Running `sudo find / -iname '*objective-see*'` reports the same four files as above.

Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: > Running `sudo find / -iname '*objective-see*'` reports the same four files as above.

What about **WhatsYourSign** in **System Preferences → Extensions**?

Comment (miccal) [inline Casks/whatsyoursign.rb]: It remains in **System Preferences → Extensions** after running both commands.

Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: Ah, I think I see the mistake. Maybe this:

```
/usr/bin/pluginkit -r ~/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync
```

If that does not work, perhaps:

```
/usr/bin/pluginkit -e ignore -i com.objective-see.WhatsYourSignExt.FinderSync
rm -rf ~/Library/Application Scripts/com.objective-see.WhatsYourSignExt.FinderSync
rm -rf ~/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync
```

Also forgot before that `/usr/bin/pluginkit -m | grep 'WhatsYourSignExt'` is another check that should be performed.

Comment (miccal) [inline Casks/whatsyoursign.rb]: Command `/usr/bin/pluginkit -r ~/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync` outputs:
```
remove: no plugin at /Users/raesy/Library/Containers/com.objective-see.WhatsYourSignExt.FinderSync
```
Very weird!

Command `/usr/bin/pluginkit -e ignore -i com.objective-see.WhatsYourSignExt.FinderSync` outputs nothing.

Running `sudo find / -iname '*objective-see*'` again reports the same four files as above.

Comment (miccal) [inline Casks/whatsyoursign.rb]: Output of command `/usr/bin/pluginkit -m | grep 'WhatsYourSignExt'`:
```
-    com.objective-see.WhatsYourSignExt.FinderSync(1.2.1)
```

Comment (miccal) [inline Casks/whatsyoursign.rb]: Output of command `/usr/bin/pluginkit -mv | grep 'WhatsYourSignExt'`:
```
-    com.objective-see.WhatsYourSignExt.FinderSync(1.2.1)	FBD17E27-E1D3-4AE5-B267-7053C0A87526	2016-12-22 13:50:16 +0000	/Applications/WhatsYourSign.app/Contents/PlugIns/WhatsYourSign.appex
```

Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: I think we’re getting closer! What about `/usr/bin/pluginkit -r /Applications/WhatsYourSign.app/Contents/PlugIns/WhatsYourSign.appex`?

Comment (miccal) [inline Casks/whatsyoursign.rb]: The command `/usr/bin/pluginkit -r /Applications/WhatsYourSign.app/Contents/PlugIns/WhatsYourSign.appex` outputs nothing, the same four files are present, BUT **WhatsYourSign** is now gone from **System Preferences → Extensions**.

Comment (miccal) [inline Casks/whatsyoursign.rb]: `/usr/bin/pluginkit -mv | grep 'WhatsYourSignExt'` outputs nothing!

Comment (vitorgalvao) [inline Casks/whatsyoursign.rb]: Woo hoo! We did it. That just means we just have to remove those extra files.

**Comment by vitorgalvao** [comment]
Should be good to go. Thank you for the help and availability, @miccal!

**Comment by miccal** [comment]
As always, you are most welcome.

**Pull request merged**